eth/v1alpha1/validator.proto

```diff
@@ -15,13 +15,120 @@ syntax = "proto3";
+    // data for proposer to sign.
+    rpc GetBlock(BlockRequest) returns (Block) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/block"
+        };
+    }
+
+    // Sends the newly signed beacon block to beacon node to be broadcasted to the rest of the network.
+    //
+    // The validator sends the newly signed beacon block to the beacon node so the beacon block can
+    // be included in the beacon chain. The beacon node is expected to validate and process the
+    // beacon block into its state.
+    rpc ProposeBlock(Block) returns (google.protobuf.Empty) {
+        option (google.api.http) = {
+            post: "/eth/v1alpha1/validator/block"
+        };
+    }
+
+    // Retrieves the latest valid attestation data to be attested on the beacon chain.
+    //
+    // The server returns the latest valid attestation data which represents the correct vote
+    // for the head of the beacon chain,
+    rpc GetAttestationData(AttestationDataRequest) returns (AttestationData) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/attestation"
+        };
+    }
+
+    // Sends the newly signed attestation to beacon node to be broadcasted to the rest of the network.
+    //
+    // The validator sends the newly signed attestation to the beacon node for the attestation to
+    // be included in the beacon chain. The beacon node is expected to validate and aggregate the
+    // attestations into the state.
+    rpc ProposeAttestation(Attestation) returns (google.protobuf.Empty) {
+        option (google.api.http) = {
+            post: "/eth/v1alpha1/validator/attestation"
+        };
+    }
+}
+
+message DutiesRequest {
+    // Epoch at which validators should perform their duties.
+    uint64 epoch = 1;
+    // Array of byte encoded BLS public keys.
+    repeated bytes public_keys = 2;
+}
+
+message DutiesResponse {
+    repeated Duty duties = 1;
+    message Duty {
+        // 96 byte BLS public key for the validator who's assigned to perform the following duty.
+        bytes public_key = 1;
+        // Slot at which a validator must attest.
+        uint64 attestation_slot = 2;
+        // Shard at which a validator must attest.
+        uint64 attestation_shard = 3;
+        // Slot at which a validator must propose on beacon chain,
+        // when returns 0, the block production is not required, meaning
+        // the validator did not get assigned to be a proposer for the
+        // input epoch.
+        uint64 block_proposal_slot = 4;
+    }
+}
+
+message BlockRequest {
+    // Slot for which the block should be proposed.
+    uint64 slot = 1;
+    // Validator's 32 byte randao reveal secret of the current epoch.
+    bytes randao_reveal = 2;
+}
+
+message AttestationDataRequest {
+    // The proof of custody bit (1 byte) reported by the requesting validator. To be used in phase 1.
+    bytes proof_of_custody_bit = 1;
```

Comment:
QQ: why is this called a bit if it's a byte?

Reply:
Discussed offline to add some details. The proof of custody is a single byte with a single bit set to 1. This is used for ??? :)